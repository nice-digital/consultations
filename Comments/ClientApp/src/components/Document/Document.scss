@use "@nice-digital/nds-core/scss/spacing";
@use "@nice-digital/nds-core/scss/utils";
@use "../../styles/variables" as *;

.document-comment-container {
  &:focus {
    outline: none;
  }
  &__commentButton {
    margin-left: utils.rem(-40px);
    position: absolute;
    @extend %unstyled-button;
    &:hover {
      cursor: pointer;
    }
    outline: -webkit-focus-ring-color red 5px
  }
  padding-left: utils.rem(spacing.$large);
  &.loading {
    opacity: 0.3;
    transition: opacity 0.1s;
  }
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin-top: 0;
    a {
      text-decoration: none;
      color: inherit;
    }
  }
  div.itemizedlist.indented,
  p.numbered-paragraph {
    margin-left: utils.rem(spacing.$xx-large);
  }
  div.itemizedlist.indented .paragraph-number,
  p.numbered-paragraph .paragraph-number {
    float: left;
    margin-left: utils.rem(-(spacing.$xx-large));
  }
  article.numbered-paragraph {
    display: flex;

    & > {
      h1, h2, h3, h4, h5, h6 {
        font-size: inherit;
        font-weight: normal;
        flex: 0 0 (utils.rem(spacing.$xx-large) - 10px);
        font-family: inherit;
        line-height: 1.6;
        padding-right: 10px;
      }
    }

    p:first-child {
        margin-top: 0;
    }
  }
  table {
    td {
      padding: utils.rem(spacing.$small);
    }
    margin-bottom: utils.rem(spacing.$medium);
  }
}

.caution {
  background-color: $colour-caution-our-version;
  border: utils.em(spacing.$xx-small) solid black;
  display: inline-block;
  padding: 	utils.rem(spacing.$small);
}

.visuallyhidden {
  position: absolute;
  overflow: hidden;
  clip: rect(0 0 0 0);
  height: 1px; width: 1px;
  margin: -1px; padding: 0; border: 0;
}

.alert {
  max-width: 66ch;
}
