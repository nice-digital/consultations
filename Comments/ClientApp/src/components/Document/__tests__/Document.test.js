import React from "react";
import { render, screen, waitForElementToBeRemoved } from "@testing-library/react";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import { MemoryRouter } from "react-router";
import { Document } from "../Document";
import ChapterData from "./Chapter.json";
import ConsultationData from "./Consultation.json";
import ConsultationQuestionsOnlyData from "./ConsultationQuestionsOnly.json";
import DocumentsData from "./Documents.json";

jest.mock("../../../context/UserContext", () => {
	return {
		UserContext: {
			Consumer: (props) => {
				return props.children({
					isAuthorised: true,
				});
			},
		},
	};
});

const fakeProps = {
	location: {},
	match: {
		params: {
			consultationId: 1,
			documentId: 1,
			chapterSlug: "introduction",
		},
	},
};

const documents = [
	{
		title: "Document One",
		documentId: 1,
		convertedDocument: false,
		href: "/guidance/one/123",
		chapters: null,
	},
	{
		title: "",
		documentId: 2,
		chapters: [{ slug: "chapter-two-slug" }],
		convertedDocument: true,
		href: "/guidance/two/123",
	},
	{
		title: "Document Three",
		documentId: 3,
		convertedDocument: false,
		href: "/guidance/three/123",
		chapters: null,
	},
	{
		title: "Document Four",
		documentId: 4,
		convertedDocument: true,
		href: "/guidance/four/123",
		chapters: [{ slug: "chapter-two-slug" }],
	},
];

it("should match snapshot with supplied data", () => {
	const mock = new MockAdapter(axios);
	let documentsPromise = new Promise(resolve => {
		mock
			.onGet("/consultations/api/Documents?consultationId=1")
			.reply(() => {
				resolve();
				return [200, DocumentsData];
			});
	});
	let consulatationPromise = new Promise(resolve => {
		mock
			.onGet("/consultations/api/Consultation?consultationId=1&isReview=false")
			.reply(() => {
				resolve();
				return [200, ConsultationData];
			});
	});
	let chapterPromise = new Promise(resolve => {
		mock
			.onGet(
				"/consultations/api/Chapter?consultationId=1&documentId=1&chapterSlug=introduction",
			)
			.reply(() => {
				resolve();
				return [200, ChapterData];
			});
	});
	const {container} = render(
		<MemoryRouter>
			<Document {...fakeProps} />
		</MemoryRouter>,
	);

	return Promise.all([
		documentsPromise,
		consulatationPromise,
		chapterPromise,
	]).then(async () => {
		const tutorialButton = await screen.findByText("Hide how to comment", { selector: "button" });
		expect(tutorialButton).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});

it("should not contain tutorial if the consultation does not support comments", () => {
	const mock = new MockAdapter(axios);
	let documentsPromise = new Promise(resolve => {
		mock
			.onGet("/consultations/api/Documents?consultationId=1")
			.reply(() => {
				resolve();
				return [200, DocumentsData];
			});
	});
	let consulatationPromise = new Promise(resolve => {
		mock
			.onGet("/consultations/api/Consultation?consultationId=1&isReview=false")
			.reply(() => {
				resolve();
				return [200, ConsultationQuestionsOnlyData];
			});
	});
	let chapterPromise = new Promise(resolve => {
		mock
			.onGet(
				"/consultations/api/Chapter?consultationId=1&documentId=1&chapterSlug=introduction",
			)
			.reply(() => {
				resolve();
				return [200, ChapterData];
			});
	});
	render(
		<MemoryRouter>
			<Document {...fakeProps} />
		</MemoryRouter>,
	);

	return Promise.all([
		documentsPromise,
		consulatationPromise,
		chapterPromise,
	]).then(async () => {
		await waitForElementToBeRemoved(() => screen.getByText("Loading...", { selector: "h1" }));
		expect(screen.queryAllByText("Hide how to comment", { selector: "button" }).length).toEqual(0);
	});
});

it("getDocumentChapterLinks method", () => {
	const document = new Document();
	expect(
		document.getDocumentChapterLinks(1, "introduction", 1, DocumentsData, "A title for the nav"),
	).toHaveProperty("links", [{"current": true, "isReactRoute": true, "label": "Introduction", "url": "/1/1/introduction"}, {"current": false, "isReactRoute": true, "label": "Patient-centred care", "url": "/1/1/patient-centred-care"}, {"current": false, "isReactRoute": true, "label": "Key priorities for implementation", "url": "/1/1/key-priorities-for-implementation"}, {"current": false, "isReactRoute": true, "label": "1 Guidance", "url": "/1/1/guidance"}, {"current": false, "isReactRoute": true, "label": "2 Notes on the scope of the guidance", "url": "/1/1/notes-on-the-scope-of-the-guidance"}, {"current": false, "isReactRoute": true, "label": "3 Implementation", "url": "/1/1/implementation"}, {"current": false, "isReactRoute": true, "label": "4 Research recommendations", "url": "/1/1/research-recommendations"}, {"current": false, "isReactRoute": true, "label": "5 Other versions of this guideline", "url": "/1/1/other-versions-of-this-guideline"}, {"current": false, "isReactRoute": true, "label": "6 Related NICE guidance", "url": "/1/1/related-nice-guidance"}, {"current": false, "isReactRoute": true, "label": "7 Updating the guideline", "url": "/1/1/updating-the-guideline"}, {"current": false, "isReactRoute": true, "label": "Appendix A: The Guideline Development Group and NICE project team", "url": "/1/1/appendix-a-the-guideline-development-group-and-nice-project-team"}, {"current": false, "isReactRoute": true, "label": "Appendix B: The Guideline Review Panel", "url": "/1/1/appendix-b-the-guideline-review-panel"}, {"current": false, "isReactRoute": true, "label": "Appendix C: The algorithm", "url": "/1/1/appendix-c-the-algorithm"}, {"current": false, "isReactRoute": true, "label": "Changes after publication", "url": "/1/1/changes-after-publication"}, {"current": false, "isReactRoute": true, "label": "About this guideline", "url": "/1/1/about-this-guideline"}]);
});

it("getDocumentLinks for Commentable documents", () => {
	const document = new Document();
	expect(
		document.getDocumentLinks(true, "This is the title", documents, 1, 1),
	).toHaveProperty("links", [
		{
			current: false,
			label: "Download Document",
			url: "/1/2/chapter-two-slug",
			isReactRoute: true,
		},
		{
			current: false,
			label: "Document Four",
			url: "/1/4/chapter-two-slug",
			isReactRoute: true,
		},
	]);
});

it("getDocumentLinks for Supporting documents", () => {
	const document = new Document();
	expect(
		document.getDocumentLinks(false, "This is the title", documents, 1, 1),
	).toHaveProperty("links", [
		{
			current: true,
			label: "Document One",
			url: "/guidance/one/123",
			isReactRoute: false,
		},
		{
			current: false,
			label: "Document Three",
			url: "/guidance/three/123",
			isReactRoute: false,
		},
	]);
});

it("should not caution that guidance is draft if the consultation is open", () => {
	const mock = new MockAdapter(axios);
	let documentsPromise = new Promise(resolve => {
		mock
			.onGet("/consultations/api/Documents?consultationId=1")
			.reply(() => {
				resolve();
				return [200, DocumentsData];
			});
	});
	let consulatationPromise = new Promise(resolve => {
		mock
			.onGet("/consultations/api/Consultation?consultationId=1&isReview=false")
			.reply(() => {
				resolve();
				return [200, ConsultationData];
			});
	});
	let chapterPromise = new Promise(resolve => {
		mock
			.onGet(
				"/consultations/api/Chapter?consultationId=1&documentId=1&chapterSlug=introduction",
			)
			.reply(() => {
				resolve();
				return [200, ChapterData];
			});
	});
	render(
		<MemoryRouter>
			<Document {...fakeProps} />
		</MemoryRouter>,
	);
	return Promise.all([
		documentsPromise,
		consulatationPromise,
		chapterPromise,
	]).then(async () => {
		const warningShown = screen.queryAllByText("The content on this page is not current guidance and is only for the purposes of the consultation process.").length > 0;
		expect(warningShown).toEqual(false);
	});
});
