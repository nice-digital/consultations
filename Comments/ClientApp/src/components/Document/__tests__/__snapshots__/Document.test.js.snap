// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ClientApp]  Document Component should match snapshot with supplied data 1`] = `
Array [
  "",
  <aside
    className="tutorial mt--0 pt--d pb--d"
  >
    <div
      className="container"
    >
      <button
        className="buttonAsLink"
        onClick={[Function]}
      >
        Hide how to comment
      </button>
      <div
        className="grid"
      >
        <div
          data-g="12 sm:7"
        >
          <span
            className="h1 pt--c"
          >
            How to comment
          </span>
          <p>
            Please view our 5-minute video on how to use the online commenting system.
          </p>
          <p>
            See how to:
          </p>
          <ul>
            <li>
              Create comments and respond to questions
            </li>
            <li>
              Add, edit and delete comments
            </li>
            <li>
              Create comments specific to sections of the document
            </li>
            <li>
              Review all your feedback before submitting your response
            </li>
          </ul>
        </div>
        <div
          data-g="12 sm:5"
        >
          <div
            className="tutorial__video"
          >
            <iframe
              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
              allowFullScreen={true}
              frameBorder="0"
              src="https://www.youtube.com/embed/DILcZkxVPCY?rel=0"
              title="YouTube video player"
            />
          </div>
        </div>
      </div>
    </div>
  </aside>,
  <div
    className="container"
  >
    <div
      className="grid"
    >
      <div
        data-g="12"
      >
        <nav
          aria-label="Breadcrumbs"
        >
          <p
            className="visually-hidden"
          >
            You are here:
          </p>
          <ol
            className="breadcrumbs"
            itemScope={true}
            itemType="http://schema.org/BreadcrumbList"
          >
            <li
              className="breadcrumbs__crumb"
              itemProp="itemListElement"
              itemScope={true}
              itemType="http://schema.org/ListItem"
              onClick={[Function]}
              role="presentation"
            >
              <a
                href="/"
                itemProp="item"
              >
                <span
                  itemProp="name"
                >
                  Home
                </span>
              </a>
              <meta
                content={0}
                itemProp="position"
              />
            </li>
            <li
              className="breadcrumbs__crumb"
              itemProp="itemListElement"
              itemScope={true}
              itemType="http://schema.org/ListItem"
              onClick={[Function]}
              role="presentation"
            >
              <a
                href="/guidance/indevelopment/gid-ta10232/consultation/"
                itemProp="item"
              >
                <span
                  itemProp="name"
                >
                  Consultation title
                </span>
              </a>
              <meta
                content={1}
                itemProp="position"
              />
            </li>
          </ol>
        </nav>
        <main>
          <div
            className="page-header"
          >
            Array [
              Array [
                <h1
                  className="page-header__heading mt--0"
                  data-qa-sel="changeable-page-header"
                >
                  Unstable angina and NSTEMI
                </h1>,
                <div
                  className="mb--d"
                >
                  <p
                    className="container container-full ml--0"
                  >
                    <span>
                      <span
                        className="tag tag--open"
                      >
                        Open for comments
                      </span>
                       Open until
                       
                      <time
                        dateTime={"2029-05-31T16:00:00.000Z"}
                      >
                        31 May 2029
                      </time>
                    </span>
                      
                    <a
                      href="/leadinformation"
                      onClick={[Function]}
                    >
                      Request commenting lead permission
                    </a>
                  </p>
                </div>,
              ],
              <div
                aria-live="polite"
                className="alert alert--info"
                role="status"
              >
                <p>
                  We have updated this service so that members of the same organisation can now collaborate on a joint online response.
                </p>
                <p>
                  Read our 
                  <a
                    href="https://www.nice.org.uk/news/blog/groups-and-organisations-can-work-together-on-public-consultations-response"
                  >
                    blog
                  </a>
                   to learn more.
                </p>
              </div>,
            ]
            <button
              className="btn btn--cta"
              data-gtm="comment-on-document-button"
              data-qa-sel="comment-on-consultation-document"
              onClick={[Function]}
            >
              Comment on this document
            </button>
          </div>
          <button
            className="screenreader-button"
            onClick={[Function]}
          >
            Skip navigation
          </button>
          <div
            className="grid"
          >
            <div
              className="navigationColumn"
              data-g="12 md:3"
            >
              <nav
                aria-label="Chapters in this document"
                className="stacked-nav"
              >
                <h2
                  className="stacked-nav__root"
                >
                  Chapters in this document
                </h2>
                <ul
                  className="stacked-nav__list"
                >
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current="page"
                      href="/1/1/introduction"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          Introduction
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/1/patient-centred-care"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          Patient-centred care
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/1/key-priorities-for-implementation"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          Key priorities for implementation
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/1/guidance"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          1 Guidance
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/1/notes-on-the-scope-of-the-guidance"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          2 Notes on the scope of the guidance
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/1/implementation"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          3 Implementation
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/1/research-recommendations"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          4 Research recommendations
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/1/other-versions-of-this-guideline"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          5 Other versions of this guideline
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/1/related-nice-guidance"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          6 Related NICE guidance
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/1/updating-the-guideline"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          7 Updating the guideline
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/1/appendix-a-the-guideline-development-group-and-nice-project-team"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          Appendix A: The Guideline Development Group and NICE project team
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/1/appendix-b-the-guideline-review-panel"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          Appendix B: The Guideline Review Panel
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/1/appendix-c-the-algorithm"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          Appendix C: The algorithm
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/1/changes-after-publication"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          Changes after publication
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/1/about-this-guideline"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          About this guideline
                        </span>
                      </span>
                    </a>
                  </li>
                </ul>
              </nav>
              <nav
                aria-label="Other consultation documents you can comment on"
                className="stacked-nav"
              >
                <h2
                  className="stacked-nav__root"
                >
                  Other consultation documents you can comment on
                </h2>
                <ul
                  className="stacked-nav__list"
                >
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/2/overview"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          Sinusitis (acute): antimicrobial prescribing
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/3/product-intros-056fc427-7484-4d78-8d55-0070aaa12061-2-2docx"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          Download Document
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/4/messaging-service-restartdocx"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          Download Document
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/5/introduction"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          Unstable angina and NSTEMI
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/6/messaging-service-restartdocx"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          Download Document
                        </span>
                      </span>
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                    data-qa-sel="nav-list-item"
                  >
                    <a
                      aria-current={null}
                      href="/1/7/supervisor-self-service-nice-guide-2018docx"
                      onClick={[Function]}
                    >
                      <span
                        className="grid"
                      >
                        <span
                          className="pl--c"
                          data-g="10"
                        >
                          Download Document
                        </span>
                      </span>
                    </a>
                  </li>
                </ul>
              </nav>
            </div>
            <div
              className="inPageNavColumn"
              data-g="12 md:3 md:push:6"
            >
              <nav
                aria-labelledby="inpagenav-title"
                className="in-page-nav"
              >
                <h2
                  className="in-page-nav__title"
                  id="inpagenav-title"
                >
                  On this page
                </h2>
                <ol
                  className="in-page-nav__list"
                  role="menubar"
                >
                  <li
                    className="in-page-nav__item"
                    onClick={[Function]}
                    role="presentation"
                  >
                    <a
                      aria-current={null}
                      href="/#introduction"
                      onClick={[Function]}
                      role="menuitem"
                    >
                      Introduction
                    </a>
                  </li>
                  <li
                    className="in-page-nav__item"
                    onClick={[Function]}
                    role="presentation"
                  >
                    <a
                      aria-current={null}
                      href="/#provision-of-information"
                      onClick={[Function]}
                      role="menuitem"
                    >
                      1.1 Provision of information
                    </a>
                  </li>
                  <li
                    className="in-page-nav__item"
                    onClick={[Function]}
                    role="presentation"
                  >
                    <a
                      aria-current={null}
                      href="/#assessment-of-a-patients-risk-of-future-adverse-cardiovascular-events"
                      onClick={[Function]}
                      role="menuitem"
                    >
                      1.2 Assessment of a patient's risk of future adverse cardiovascular events
                    </a>
                  </li>
                  <li
                    className="in-page-nav__item"
                    onClick={[Function]}
                    role="presentation"
                  >
                    <a
                      aria-current={null}
                      href="/#antiplatelet-therapy"
                      onClick={[Function]}
                      role="menuitem"
                    >
                      1.3 Antiplatelet therapy
                    </a>
                  </li>
                  <li
                    className="in-page-nav__item"
                    onClick={[Function]}
                    role="presentation"
                  >
                    <a
                      aria-current={null}
                      href="/#antithrombin-therapy"
                      onClick={[Function]}
                      role="menuitem"
                    >
                      1.4 Antithrombin therapy
                    </a>
                  </li>
                  <li
                    className="in-page-nav__item"
                    onClick={[Function]}
                    role="presentation"
                  >
                    <a
                      aria-current={null}
                      href="/#management-strategies"
                      onClick={[Function]}
                      role="menuitem"
                    >
                      1.5 Management strategies
                    </a>
                  </li>
                </ol>
              </nav>
            </div>
            <div
              className="documentColumn"
              data-g="12 md:6 md:pull:3"
            >
              <div
                aria-live="polite"
                className="document-comment-container "
                id="content-start"
              >
                <div
                  onMouseUp={[Function]}
                  role="presentation"
                >
                  <div
                    className="selection-container unselectable"
                    style={
                      Object {
                        "display": "none",
                        "left": undefined,
                        "top": undefined,
                      }
                    }
                  >
                    <button
                      className="btn"
                      onClick={[Function]}
                    >
                      <span
                        aria-hidden="true"
                        className="icon icon--comment unselectable"
                      />
                        Comment
                    </button>
                  </div>
                  <div
                    className="chapter"
                    title="Introduction"
                    xmlns="http://www.w3.org/1999/xhtml"
                  >
                    <h2
                      className="title"
                    >
                      <button
                        className="document-comment-container__commentButton"
                        data-gtm="button"
                        data-gtm-action="Clicked"
                        data-gtm-category="Consultation comments page"
                        data-gtm-label="Comment on chapter"
                        data-qa-sel="in-text-comment-button"
                        data-sectionnumber={null}
                        onClick={[Function]}
                        tabIndex={0}
                        title="Comment on chapter"
                      >
                        <span
                          aria-hidden="true"
                          className="icon icon--comment"
                        />
                        <span
                          className="visually-hidden"
                        >
                          Comment on 
                          chapter
                          : 
                          Introduction 
                        </span>
                      </button>
                      <a
                        className="annotator-chapter"
                        data-heading-type="chapter"
                        id="introduction"
                        style={
                          Object {
                            "position": "relative",
                          }
                        }
                        title=""
                        xmlns=""
                      >
                        Introduction 
                        <span
                          className="annotator-adder"
                        />
                      </a>
                    </h2>
                    <div
                      className="informaltable"
                    >
                      <table
                        border="1"
                      >
                        <colgroup>
                          <col />
                        </colgroup>
                        <tbody>
                          <tr>
                            <td
                              align="left"
                              valign="top"
                            >
                              <p>
                                This guideline updates and replaces recommendations for the early management of unstable angina and NSTEMI from NICE technology appraisal guidance 
                                <a
                                  className="link"
                                  href="http://guidance.nice.org.uk/TA47"
                                  target="_top"
                                >
                                  47
                                </a>
                                 and 
                                <a
                                  className="link"
                                  href="http://guidance.nice.org.uk/TA80"
                                  target="_top"
                                >
                                  80
                                </a>
                                .
                              </p>
                              <p>
                                Recommendation 1.3.6 has been replaced by recommendation 1.3.18 in 
                                <a
                                  className="link"
                                  href="http://guidance.nice.org.uk/cg172"
                                  target="_top"
                                >
                                  MI – secondary prevention: Secondary prevention in primary and secondary care for patients following a myocardial infarction
                                </a>
                                .
                              </p>
                              <p>
                                Recommendation 1.5.11 has been updated to take into account people with a learning disability.
                              </p>
                              <p>
                                See 
                                <a
                                  className="link"
                                  href="http://publications.nice.org.uk/unstable-angina-and-nstemi-cg94/changes-after-publication"
                                  target="_top"
                                >
                                  Changes after publication
                                </a>
                                 for details.
                              </p>
                            </td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                    <p>
                      The term 'acute coronary syndromes' encompasses a range of conditions from unstable angina to ST-segment-elevation myocardial infarction (STEMI), arising from thrombus formation on an atheromatous plaque. This guideline addresses the early management of unstable angina and non-ST-segment-elevation myocardial infarction (NSTEMI) once a firm diagnosis has been made and before discharge from hospital. If untreated, the prognosis is poor and mortality high, particularly in people who have had myocardial damage. Appropriate triage, risk assessment and timely use of acute pharmacological or invasive interventions are critical for the prevention of future adverse cardiovascular events (myocardial infarction, stroke, repeat revascularisation or death). The guideline does not cover the management of STEMI or specific complications of unstable angina and NSTEMI such as cardiac arrest or acute heart failure. Assessment and classification of people presenting with undifferentiated chest pain are covered in 'Chest pain of recent onset' (NICE clinical guideline 95)
                      <sup>
                        [
                        <a
                          aria-current={null}
                          className="footnote"
                          data-identified-as="internal-link"
                          href="/#ftn.footnote_1"
                          id="footnote_1"
                          name="footnote_1"
                          onClick={[Function]}
                          rel="internal"
                          tabIndex="0"
                        >
                          1
                        </a>
                        ]
                      </sup>
                      .
                    </p>
                    <p>
                      The guideline will assume that prescribers will use a drug's summary of product characteristics to inform decisions made with individual patients.
                    </p>
                    <p>
                      This guideline recommends some drugs for indications for which they do not have a UK marketing authorisation at the date of publication, if there is good evidence to support that use. Unlicensed or off-label use is indicated by a footnote. 
                    </p>
                    <p>
                      Throughout the guideline, the term 'angiography' refers to invasive angiography.
                    </p>
                    <p>
                      Recommendations 1.3.4 to 1.3.8 update and replace recommendations for the early management of unstable angina and NSTEMI from '
                      <a
                        className="link"
                        href="http://guidance.nice.org.uk/TA80"
                        target="_top"
                      >
                        Clopidogrel in the treatment of non-ST-segment-elevation acute coronary syndrome
                      </a>
                      ', NICE technology appraisal guidance 80 (TA 80)
                    </p>
                    <p>
                      Recommendations 1.3.9 to 1.3.11 update and replace recommendations for the early management of unstable angina and NSTEMI from '
                      <a
                        className="link"
                        href="http://guidance.nice.org.uk/TA47"
                        target="_top"
                      >
                        Guidance on the use of glycoprotein IIb/IIIa inhibitors in the treatment of acute coronary syndrome
                      </a>
                      ', NICE technology appraisal guidance 47 (TA 47).
                    </p>
                    <div
                      className="footnotes"
                    >
                      <br />
                      <hr
                        align="left"
                        width="100"
                      />
                      <div
                        className="footnote"
                      >
                        <p>
                          <sup>
                            [
                            <a
                              aria-current={null}
                              className="para"
                              data-identified-as="internal-link"
                              href="/#footnote_1"
                              id="ftn.footnote_1"
                              name="ftn.footnote_1"
                              onClick={[Function]}
                              rel="internal"
                              tabIndex="0"
                            >
                              1
                            </a>
                            ] 
                          </sup>
                           More information on 'Chest pain of recent onset' (
                          <a
                            className="link"
                            href="http://www.nice.org.uk/guidance/CG95"
                            target="_top"
                          >
                            NICE clinical guideline 95
                          </a>
                          ) is available. 
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </main>
      </div>
    </div>
  </div>,
]
`;
